pkg/compose/docker.go

```diff
@@ -0,0 +1,55 @@
+package compose
+
+import (
+	"context"
+	"errors"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+)
+
+// Docker is the implementation of the Docker container engine
+type Docker struct {
+	client *client.Client
+}
+
+// RunContainer will run a docker container
+func (engine Docker) RunContainer(image string) error {
+	containerID, err := engine.GetContainer(image)
+	if err != nil {
+		return err
+	}
+	ctx := context.Background()
+	if err := engine.client.ContainerStart(ctx, containerID, types.ContainerStartOptions{}); err != nil {
+		return err
+	}
+	return nil
+}
+
+// ActiveContainers returns the list of running docker containers
+func (engine Docker) ActiveContainers() (int, error) {
+	count := 0
+
+	containers, err := engine.client.ContainerList(context.Background(), types.ContainerListOptions{})
+	if err != nil {
+		return count, err
+	}
+	for i := 0; i < len(containers); i++ {
+		count++
+	}
+	return count, nil
+}
+
+// GetContainer returns a container id from the containers name
+func (engine Docker) GetContainer(image string) (string, error) {
```

Comment:
Gotcha. Also, internal tests are done by just putting the tests in the same package so they have access to the internal functions